tests/sub_group/sub_group_api.cpp

```diff
@@ -170,6 +217,18 @@ TEST_CASE("sub-group api", "[sub_group]") {
   // if possible, reduce size by one to attempt to create incomplete sub-groups
   if (local_size_dim > 1) {
     --local_size_dim;
+    // if required size for result buffer for current work group dimension size
+    // exceed maximum memory allocation size on device reduce initial
+    // local_size_dim (i.e. current local_size_dim + 1) to nearest multiple two
+    // and if possible, again reduce size by one to attempt to create incomplete
+    // sub-groups
+    size_t required_res_buffer_size =
+        result_buf_size(make_global_range(local_range, local_size_dim).size());
+    if (result_buffer_cant_be_alloc_in_device(required_res_buffer_size)) {
+      local_size_dim = reduce_work_group_size_to_suitable_multiple_two(
+          local_range, local_size_dim + 1);
+      local_size_dim = local_size_dim > 1 ? local_size_dim - 1 : local_size_dim;
```

Comment:
Unclear. Probably less obfuscated:
```suggestion
      local_size_dim = std::max(local_size_dim - 1, 1);
```